pkg/policy/selectorcache.go

```diff
@@ -705,21 +704,29 @@ func (sc *SelectorCache) AddIdentitySelector(user CachedSelectionUser, selector
 	return newIDSel, true
 }
 
-func (sc *SelectorCache) removeSelectorLocked(selector CachedSelector, user CachedSelectionUser) {
+func (sc *SelectorCache) removeSelectorLocked(selector CachedSelector, user CachedSelectionUser) func() {
 	key := selector.String()
 	sel, exists := sc.selectors[key]
 	if exists {
-		if sel.removeUser(user) {
+		last, cb := sel.removeUser(user)
+		if last {
 			delete(sc.selectors, key)
 		}
+		return cb
 	}
+	return nil
 }
 
 // RemoveSelector removes CachedSelector for the user.
 func (sc *SelectorCache) RemoveSelector(selector CachedSelector, user CachedSelectionUser) {
 	sc.mutex.Lock()
-	sc.removeSelectorLocked(selector, user)
+	cb := sc.removeSelectorLocked(selector, user)
```

Comment:
This needs to be wrapped in a `func()` doesn't it? Otherwise cb is the return of `removeSelectorLocked` but that returns nothing! (it's the none return? void?)